### PR TITLE
Bump version to 2024.2

### PR DESCRIPTION
**Pull request by prusse-martin**

### CODE REVIEW

**Review by cauebs — APPROVED** [review]

File: src/alfasim_sdk/_internal/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "2024.1"
+__version__ = "2024.2"
```
Comment (nicoddemus) [inline src/alfasim_sdk/_internal/version.py:1]: How about using a `.dev` suffix to signal this has not been released yet?

```suggestion
__version__ = "2024.2.dev"
```

**Review by nicoddemus — APPROVED** [review]

**Review by ro-oliveira95 — APPROVED** [review]

File: src/alfasim_sdk/_internal/version.py (hunk 1 above)
Comment (prusse-martin) [reply src/alfasim_sdk/_internal/version.py:1]: OK... I will update the alfasim release process to mention we should update the sdk version when starting an alfasim RB.

**Comment by prusse-martin** [comment]
@nicoddemus 
From alfasim build after the introduction of `.dev` in version:
```
E               AssertionError: Error messages not found in expected:
E                   "Unsupported version of "hydrates" plugin, current ALFAsim version uses alfasim-sdk="2024.2.dev0" and plugin has the following requirement=">=2024.2""
```
The version spec used when creating the packages is `">=MAJOR.MINOR"` and `MAJOR.MINOR.devX < MAJOR.MINOR` I will add the full version on packages from now on.
I could drop the `.devX` part when comparing in alfasim but this looks safer.

**Comment by nicoddemus** [comment]
> The version spec used when creating the packages is ">=MAJOR.MINOR" and MAJOR.MINOR.devX < MAJOR.MINOR I will add the full version on packages from now on.

Why is a plugin depending on `>=2024.2` which has not even been released yet? Shouldn't it depend on `>=2024.1`, which is the latest release?

**Comment by prusse-martin** [comment]
> Why is a plugin depending on >=2024.2 which has not even been released yet? Shouldn't it depend on >=2024.1, which is the latest release?

Because we did just release `2024.1`, plugins build with that version will depend on `2024.1`, sdk just did enter the development phase of `2024.2`.

**Review by BeneBr — APPROVED** [review]

**Pull request merged**